server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorFactory.java

```diff
@@ -59,18 +62,141 @@
     private final TermsAggregator.BucketCountThresholds bucketCountThresholds;
     private final boolean showTermDocCountError;
 
+    // TODO: Registration should happen on the actual aggregator classes
```

Comment:
TODO: delete this TODO.  We've come to like the factory-does-the-registration pattern just fine.